File: C++/CommonStock.cpp
/*
 * CommonStock.cpp
 *
 *  Created on: 28/02/2016
 *      Author: oscar
 */

#include "CommonStock.h"

CommonStock::CommonStock(Symbol symbol, unsigned lastDividend, unsigned parValue) :
        Stock(symbol, lastDividend, parValue)
{
}

CommonStock::~CommonStock()
{
}

double
CommonStock::getDividend() const
{
    return m_lastDividend;
}

File: Python/CommonStock.py
from Stock import Stock

class CommonStock(Stock):
    """Common stock"""
    def __init__(self, symbol, lastDividend, parValue):
        Stock.__init__(self, symbol, lastDividend, parValue)
        
    def getDividendYield(self, price):
        return self.lastDividend / price

        
